.github/workflows/test-runner-nightly-integration.yml

```diff
@@ -0,0 +1,101 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Test Runner Nightly Integration Tests
+
+on:
+  push
```

Comment:
Without more qualification, won't this run on any push to any branch in the repository?
Or is that controlled elsewhere?

Reply:
I updated the workflow to be triggered on push to `dev` branch or PRs.